Automattic-Tracks-iOS/Extensions/FileManager+Extensions.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+extension FileManager {
+
+    var documentsDirectory: URL {
+        let documentsDirectory = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first!
+        return URL(fileURLWithPath: documentsDirectory, isDirectory: true)
+    }
+
+    func contentsOfDirectory(at url: URL) throws -> [URL] {
+        return try FileManager.default.contentsOfDirectory(atPath: url.path).map { url.appendingPathComponent($0) }
+    }
+
+    func contents(atUrl url: URL) -> Data? {
+        return self.contents(atPath: url.path)
+    }
+
+    func fileExistsAtURL(_ url: URL) -> Bool {
+        return FileManager.default.fileExists(atPath: url.path)
+    }
```

Comment:
I realize this is a very common thing in iOS land, but I am still failing to see the value in having these extensions instead of directly calling them in the code. Isn't this adding more cognitive load to whoever is looking at this code because they now also need to look into how it's implemented?

Anyway, this is not really a request to change or even a nitpicky comment, just curious about why this is the common approach.

Reply:
Good question – by default, `FileManager` likes to work with paths, which are just a `String`.  Unfortunately, it's _really_ easy to mess up (NS)URL => String conversion (do I use `absoluteString`? `relativeString`? `path`? `relativePath`?)  – there's a _lot_ of properties on `URL`, so not having to think about these conversions at the call sites, and being able to perform operations on `FileManager` in a type-safe way like this (just always represent paths as URLs and you can never do it wrong) is pretty beneficial, even if it has some silly aliases like this.

Reply:
That makes a lot of sense, thanks a lot for explaining all this to me!